Sources/NIOFileSystem/Docs.docc/NIOFileSystem.md

```diff
@@ -0,0 +1,174 @@
+# ``NIOFileSystem``
+
+A file system library for Swift.
+
+## Overview
+
+This module implements a file system library for Swift, providing ways to interact with and manage
+files. It provides a concrete ``FileSystem`` for interacting with the local file system in addition
+to a set of protocols for creating other file system implementations.
+
+``NIOFileSystem`` is cross-platform with the following caveats:
+- _Platforms don't have feature parity or system-level API parity._ Where this is the case these
+  implementation details are documented. One example is copying files, on Apple platforms files are
+  cloned if possible.
+- _Features may be disabled on some systems._ One example is extended attributes.
+- _Some types have platform specific representations._ These include the following:
+  - File paths on Apple platforms and Linux (e.g. `"/Users/hal9000/"`) are different to paths on
+    Windows (`"C:\Users\hal9000"`).
+  - Information about files is different on different platforms. See ``FileInfo`` for further
+    details.
+
+## A Brief Tour
+
+The following sample code demonstrates a number of the APIs offered by this module:
+
+```swift
+import NIOFileSystem
+
+// NIOFileSystem provides access to the local file system via the FileSystem
+// type which is available as a global shared instance.
+let fileSystem = FileSystem.shared
+
+// Files can be inspected by using 'info':
+if let info = try await fileSystem.info(forFileAt: "/Users/hal9000/demise-of-dave.txt") {
+  print("demise-of-dave.txt has type '\(info.type)'")
+} else {
+  print("demise-of-dave.txt doesn't exist")
+}
+
+// Let's find out what's in that file.
+do {
+  // Reading a whole file requires a limit. If the file is larger than the limit
+  // then an error is thrown. This avoids accidentally consuming too much memory
+  // if the file is larger than expected.
+  let plan = try await Array<UInt8>(
+    contentsOf: "/Users/hal9000/demise-of-dave.txt",
+    maximumBytesAllowed: 1024 * 1024
```

Comment:
Could we use the `ByteCount` type here instead to showcase it?

Reply:
Yeah good catch, this API no longer exists.